opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -826,10 +809,12 @@ def __init__(
         active_span_processor: Union[
             SynchronousMultiSpanProcessor, ConcurrentMultiSpanProcessor
         ] = None,
+        ids_generator: trace_api.IdsGenerator = RandomIdsGenerator()
```

Comment:
You should have the default value as `None` and then instantiate it in the constructor.

https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

Reply:
Wow I never knew this, thanks for pointing it out!